CONTRIBUTING.md

```diff
@@ -77,6 +78,60 @@ to do this is to invoke `rustc` like this:
 $ RUST_BACKTRACE=1 rustc ...
 ```
 
+## The Build System
+
+The build system for Rust is complex. It covers bootstrapping the compiler,
+running tests, building documentation and more. Unless you are familiar with
+Makefiles, I wouldn't suggest trying to understand everything going on in
```

Comment:
* "I" rubs me the wrong way, and above (line 45) "we" is used.
* I wouldn't suggest *anyone* to try and understand everything about the build system, unless they specifically want to improve it (and then, the future-proof way to do that is to improve on #31123 rather than fiddling with the Makefiles).